js/customize-snapshots-admin.js

```diff
@@ -0,0 +1,73 @@
+( function( $ ) {
+	'use strict';
+
+	$( function() {
+		var $link, linkText, linkActions, dataSlug, initializeLink;
+
+		$link = $( '.snapshot-toggle-setting-removal' );
+		linkText = [ 'Remove setting', 'Restore setting' ];
+		linkActions = [ 'remove', 'restore' ];
+		dataSlug = 'cs-action';
+
+		initializeLink = function() {
+			$link.text( linkText[ 0 ] )
+				.data( dataSlug, linkActions[ 0 ] );
+		};
+
+		initializeLink();
+
+		$link.on( 'click', function( event ) {
```

Comment:
The function body should be refactored, to make the logic for the functions to not rely on `this` but be passed the context as an explicit argument. So `hideSettingAndChangeLinkText` can take an argument for the setting `details` jQuery object, for example, instead of relying on the inherited `this` context.